java-checks/src/main/java/org/sonar/java/checks/DeprecatedHashAlgorithmCheck.java

```diff
@@ -90,27 +141,27 @@
   @Override
   protected void onMethodInvocationFound(MethodInvocationTree mit) {
     String methodName = MethodsHelper.methodName(mit).name();
-    String algorithm = ALGORITHM_BY_METHOD_NAME.get(methodName);
+    InsecureAlgorithm algorithm = ALGORITHM_BY_METHOD_NAME.get(methodName);
     if (algorithm == null) {
       algorithm = algorithm(mit.arguments().get(0));
     }
-    boolean isMd5 = MD5.equalsIgnoreCase(algorithm);
-    boolean isSha1 = SHA1.equalsIgnoreCase(algorithm);
-    if (isMd5 || isSha1) {
-      String msgAlgo = isSha1 ? "SHA-1" : algorithm;
-      reportIssue(MethodsHelper.methodName(mit), "Use a stronger hashing algorithm than " + msgAlgo + ".");
+    if (algorithm != null) {
+      reportIssue(MethodsHelper.methodName(mit), "Use a stronger hashing algorithm than " + algorithm.toString() + ".");
     }
   }
 
-  private static String algorithm(ExpressionTree invocationArgument) {
+  @CheckForNull
+  private static InsecureAlgorithm algorithm(ExpressionTree invocationArgument) {
     ExpressionTree expectedAlgorithm = invocationArgument;
     ExpressionTree defaultPropertyValue = JavaPropertiesHelper.retrievedPropertyDefaultValue(invocationArgument);
     if (defaultPropertyValue != null) {
       expectedAlgorithm = defaultPropertyValue;
     }
     if (expectedAlgorithm.is(Tree.Kind.STRING_LITERAL)) {
-      String algo = LiteralUtils.trimQuotes(((LiteralTree) expectedAlgorithm).value());
-      return algo.replaceAll("-", "");
+      String algorithmName = LiteralUtils.trimQuotes(((LiteralTree) expectedAlgorithm).value());
+      return Arrays.stream(InsecureAlgorithm.values())
+        .filter(alg -> alg.match(algorithmName))
+        .findFirst().orElse(null);
```

Comment:
Maybe change the method to return an `Optional<InsecureAlgorithm>` rather than mixing null and stream api?